tests/run/cpp_classes_def.pyx

```diff
@@ -269,3 +269,7 @@ def test_CppClassWithDocstring():
     cdef CppClassWithDocstring *c = new CppClassWithDocstring()
     del c
     print "OK"
+
+_WARNINGS="""
+24:4: Unraisable exception in function 'RegularPolygon.area'.
```

Comment:
This comes from `Symtab.Scope.declare()`, more precisely line 515 `alt_entry.pos = pos`.
`RegularPolygon` inherits from `Shape`. So `RegularPolygon.area()` is an inherited method which is redefined.

Before this PR changes, the entry corresponding to `RegularPolygon.area()` had a `pos` attribute equal to `None`,
because inherited methods are declared this way (inherited methods are inside the `inherited_var_entries` attribute).

With this PR, the inherited method is still declared with `pos = None`, but we update `pos` as soon as we have a redefinition of `RegularPolygon.area()`.

In other words, we changed from:
```
Unraisable exception in function 'RegularPolygon.area'.
```
to:
```
24:4: Unraisable exception in function 'RegularPolygon.area'.
```

Now, why does this change the test suite results ?
Because warnings without positions are discarded by the test suite.
I guess the exact place where it's done is through the necessary position groups in the
`match_error` regex of `ErrorWriter` class in `runtests.py`. (line 607)
Warnings with no position aren't matched by this regex (if I understand it correctly), so they are silently discarded.